assets/src/edit-story/components/movable/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import Moveable from 'react-moveable';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+
+const CORNER_HANDLES = [ 'nw', 'ne', 'sw', 'se' ];
+const ALL_HANDLES = [ 'n', 's', 'e', 'w', 'nw', 'ne', 'sw', 'se' ];
+
+function Movable( {
+	rotationAngle,
+	x,
+	y,
+	type,
+	targetEl,
+	pushEvent,
+	setPushEvent,
+} ) {
+	const moveable = useRef();
+
+	const {
+		actions: { setPropertiesOnSelectedElements },
+	} = useStory();
+
+	useEffect( () => {
+		if ( moveable.current ) {
+			// If we have persistent event then let's use that.
+			if ( pushEvent ) {
+				moveable.current.moveable.dragStart( pushEvent );
+				// Remove the event right after using, not to influence other elements.
+				setPushEvent( null );
```

Comment:
We can avoid passing `setPushEvent` down and forcing re-rendering by either:

1. We don't have to reset it. We can simply check above that `pushEvent.currentTarget == targetEl` and it's a recent event.
2. We could instead pass `pushEventRef` as a ref and use `pushEventRef.current` and clear it out w/o triggering re-rendering.
3. We can keep the re-rendering. It's probably not very expensive :) But might be worth looking into it.

Reply:
We should probably do the check for the target elements matching even if we do reset the `pushEvent` afterward :)

On resetting the `pushEvent` -- it feels cleaner not to leave an old event persisted after it's purpose has been fulfilled. However, I guess it shouldn't cause any side effects if we'll always check that the target is correct.

Will go with the option 1. and remove resetting the event for now.